docs/src/tutorial.md

```diff
@@ -0,0 +1,301 @@
+In fact, type issues from dependencies are often so plentiful they flood your analysis with false positives, which can make working with JET harder.
+
+To reduce false positives, you can use the keywords `ignored_modules` and `target_modules`.Both take an iterable of modules.
+
+The former removes any errors that originate from any of the given modules, while the latter removes any errors _except_ ones originating from these modules.
+
+For example, in the REPL (which is in module `Main`), we can define:
+```julia
+f(x) = first(x) + 1
+```
+
+This throws in a `Base` function if we pass `nothing` into it:
+```julia
+julia> @report_call f(nothing)
+═════ 1 possible error found ═════
+┌ @ REPL[68]:1 first(x)
+│┌ @ abstractarray.jl:424 x = iterate(itr)
+││ no matching method found `iterate(::Nothing)`: x = iterate(itr::Nothing)
+│└────────────────────────
+```
+
+Since the error originates from `Base`, we can filter the error away by ignoring `Base`, or equivalently, we may retain only the ones from `Main`. Note that we pass `(Base,)` as a 1-element Tuple of modules:
+```
+julia> @report_call ignored_modules=(Base,) f(nothing)
+No errors detected
+
+julia> @report_call target_modules=(Main,) f(nothing)
+No errors detected
+```
+
+The `AnyFrameModule` construct can be used to filter for (or against) any error where _any_ of the function calls in the callchain originates from the given module.
+For example, in the example above, the function call begins in `Main` and ends in `Base`, so the callchain includes both modules.
+Ignoring `AnyFrameModule(Base)` _or_ `AnyFrameModule(Main)` will then ignore the error:
+
+```julia
+julia> @report_call ignored_modules=(AnyFrameModule(Base),) f(nothing)
+No errors detected
+
+julia> @report_call ignored_modules=(AnyFrameModule(Main),) f(nothing)
+No errors detected
+```
+
+Similarly, the error would be retained if `target_modules` would have been `AnyFrameModule(Base)` _or_ `AnyFrameModule(Main)`.
+
+Beware that this filtering may filter away legitimate problems in your package. In the example above, if your code calls `f(nothing)`, the error may originate from Base, but it's clearly an error in your own code to call `f(nothing)`.
+So it is recommended to only filter away modules if they produce so many false positives that it makes using JET difficult.
+
+#### Analyze scripts and apps by using a `main` function
+Scripts and apps called from the command line have a single logical entry point. If you wrap the logic in a `main` function, JET can analyse the entire script.
+
+For example, suppose you made this command-line script which added two numbers from the command line:
+
+```julia
+a = parse(Int, first(ARGS))
+b = parse(Int, last(ARGS))
+println(a + b)
+```
+
+You could rewrite this as:
+```julia
+function main()
+    a = parse(Int, first(ARGS))
+    b = parse(Int, last(ARGS))
+    println(a + b)
+end
+
+main()
+```
+
+, and then analyse the script with JET using `@report_call main()`.
```

Comment:
Or alternatively, you can use `report_file("main.jl")` for this sort of use case.